packages/plugin-javascript/tests/utilities.test.ts

```diff
@@ -0,0 +1,351 @@
+
+        await writeToSrc(workspace, 'index.js');
+        const indexModifiedTime = getModifiedTime(
+          testPackage.fs.resolvePath('src', 'index.js'),
+        );
+
+        await writeToSrc(workspace, 'file-a.ts');
+        const fileAModifiedTime = getModifiedTime(
+          testPackage.fs.resolvePath('src', 'file-a.ts'),
+        );
+
+        await writeToSrc(workspace, 'file-b.mjs');
+        const fileBModifiedTime = getModifiedTime(
+          testPackage.fs.resolvePath('src', 'file-b.mjs'),
+        );
+
+        const latestModifiedTime = getLatestModifiedTime(
+          testPackage,
+          fileExtensions,
+        );
+
+        expect(latestModifiedTime).toBeGreaterThan(indexModifiedTime);
+        expect(latestModifiedTime).toBeGreaterThan(fileAModifiedTime);
+        expect(latestModifiedTime).toEqual(fileBModifiedTime);
+      });
+    });
+
+    it('excludes files with extensions not included in the provided list', async () => {
+      await withWorkspace('simple-package', async (workspace) => {
+        const testPackage = new Package({
+          name: 'simple-package',
+          root: workspace.root,
+        });
+        const fileExtensions = ['.js'];
+
+        await writeToSrc(workspace, 'index.js');
+
+        const latestModifiedTime = getLatestModifiedTime(
+          testPackage,
+          fileExtensions,
+        );
+
+        await writeToSrc(workspace, 'typescript-file.ts');
+
+        const updatedLatestModifiedTime = getLatestModifiedTime(
+          testPackage,
+          fileExtensions,
+        );
+
+        expect(updatedLatestModifiedTime).toEqual(latestModifiedTime);
+      });
+    });
+  });
+
+  describe('generateBabelPackageCacheValue()', () => {
+    const options = {
+      babelConfig: {
+        presets: ['@babel/some-preset', '@babel/some-other-preset'],
+        plugins: ['@babel/some-plugin', '@babel/some-other-plugin'],
+      },
+      outputPath: 'build/esm',
+      extension: '.mjs',
+      exportStyle: ExportStyle.EsModules,
+      babelCacheDependencies: [
+        '@babel/some-plugin',
+        '@babel/some-other-plugin',
+      ],
+      babelIgnorePatterns: ['.json', '.graphql'],
+      babelExtensions: ['.js', '.mjs', '.ts'],
+    };
+
+    it('generates the same hash for unchanged options/dependencies/modified time', async () => {
+      await withWorkspace('simple-package', async (workspace) => {
+        const testPackage = new Package({
+          name: 'simple-package',
+          root: workspace.root,
+        });
+
+        await writeToSrc(workspace, 'index.js');
+        await writeToSrc(workspace, 'file-a.js');
+        await writeToSrc(workspace, 'file-b.js');
+
+        const hash1 = generateBabelPackageCacheValue(testPackage, options);
+        const hash2 = generateBabelPackageCacheValue(testPackage, options);
+
+        expect(hash1).toEqual(hash2);
+      });
+    });
+
+    it('generates a different hash if the Babel config changes', async () => {
```

Comment:
There's a lot of duplication across these `generates a different hash if .... changes` tests. Can we reduce all this bloat?

Some ideas: 

* Create a module-level utility so you can do a one-liner for `const testPackage = createTestPackage('simple-package, workspace.root)`
* Do we need the `writeToSrc` calls at all since we just care about the options passed in? If we need 'em lets stick to just one call. Unless I'm missing something here and we need all 3.


Something like, perhaps? 

```tsx
      await withWorkspace('simple-package', async (workspace) => {
        const testPackage = createTestPackage('some-package', workspace.root);
       // add writeToSrc calls if we need 'em

        const newOptions = {
          ...options,
          babelConfig: {
            ...options.babelConfig,
            plugins: [...options.babelConfig.plugins, '@babel/some-new-plugin'],
          },
        };

        expect(
            generateBabelPackageCacheValue(testPackage, options)
         ).not.toEqual(
            generateBabelPackageCacheValue(testPackage, newOptions)
         );
      });
```